First iteration join form

Dan-Sherman opened the pull request.

Dan-Sherman committed:
First iteration join form

Dan-Sherman committed:
Finishing Form Functionality

Finished adding in functionality to the form

Dan-Sherman committed:
Fixed text below About Yourself textarea

Text was duplicated from Game Experience, amended now

TomBurch commented on Archhub:
```diff
@@ -0,0 +1,7 @@
+-----BEGIN OPENSSH PRIVATE KEY-----
```
You're going to want to regenerate your SSH keys - try not to upload in the future 😅

Dan-Sherman replied:
I don't think this was actually ever used, am I okay to delete this file? I think this was me generating it yesterday but this was an incorrect one...

TomBurch commented on resources/views/pages/Hub/Missions/Missions.svelte:
```diff
@@ -82,7 +82,7 @@
     />
     <MissionCollection
-        title={"Played missions"}
+        title={"arcomm"}
```
Change left over from testing

Dan-Sherman replied:
reverted change on this back to the previous line

TomBurch commented on resources/views/pages/Public/App.svelte:
```diff
@@ -7,4 +7,6 @@
 
-<div class="grid h-screen-no-nav place-items-center border border-gray-200 border-gray-700 bg-gray-800 p-3 shadow-md" />
+<div class="grid h-screen-no-nav place-items-center border border-gray-200 border-gray-700 bg-gray-800 p-3 shadow-md">
+  <h1>Hello World</h1>
```
Change left over from testing

Dan-Sherman replied:
Reverted change to previous line on this

TomBurch commented on resources/views/pages/Public/Join.svelte:
```diff
@@ -8,98 +8,135 @@
+          <input
+              class="mb-3 block w-full appearance-none rounded border border-gray-200 bg-gray-200 py-3 px-4 leading-tight text-gray-700 focus:border-gray-500 focus:bg-white focus:outline-none invalid:border-red-500"
+              id="grid-location"
+              placeholder="Make it as long and as crazy as you'd like."
+              required
+          />
+          <p class="text-xs italic text-gray-400">You can be as specific as you like</p>
```
I don't think we need both the placeholder and text below to give the same message - maybe change placeholder to `Location`?

Dan-Sherman replied:
Placeholder amended to Location

TomBurch commented on resources/views/pages/Public/Join.svelte:
```diff
@@ -8,98 +8,135 @@
+            class="mb-3 block w-full appearance-none rounded border border-gray-200 bg-gray-200 py-3 px-4 leading-tight text-gray-700 focus:border-gray-500 focus:bg-white focus:outline-none invalid:border-red-500"
+            id="grid-discord"
+            type="text"
+            placeholder="Your Discord Username."
```
Some placeholders end with a `.` and some don't, to be consistent I think we should remove the `.`

Dan-Sherman replied:
Fixed this, should all be consistent from next push

TomBurch commented on resources/views/pages/Public/Join.svelte:
```diff
@@ -8,98 +8,135 @@
+                  <option>Yes</option>
+              </select>
+          </div>
+            <p class="mt-2 text-xs italic text-gray-400">
+              Arma - Saturday at 1800 Zulu
+              DCS - Sunday at 1800 Zulu
+            </p>
```
We should keep these on separate lines for legibility

Dan-Sherman replied:
Separated on next push

TomBurch commented on resources/views/pages/Public/Join.svelte:
```diff
@@ -8,98 +8,135 @@
+    </p>
+  </div>
+  <hr class="h-px my-8 bg-gray-200">
+  <div class="-mx-3 mb-6 flex flex-wrap">
+      <div class="mb-6 w-full px-3 md:mb-0 md:w-1/2">
+          <label class="mb-2 block text-xs font-bold uppercase tracking-wide text-gray-200" for="grid-name">
```
There's a lot of repeated styling used for each input, we should consider how it can be simplified

Dan-Sherman replied:
I think I'd need time to work on this sort of thing, it's whether this is a now thing or something I revisit post main page completion?

Dan-Sherman committed:
Responding to Comments

Removed SSH files
Reverted Testing changes
Placeholder on Location amended
Consistent styling on placeholders
Separated text on OP Times

TomBurch commented on resources/views/pages/Public/Join.svelte:
```diff
@@ -8,98 +8,136 @@
-                type="text"
-            />
-        </div>
+<form class="h-screen-no-nav border border-gray-700 bg-gray-800 p-3 shadow-md" id="join-form">
```
Replacing `h-screen-no-nav` with `min-h-screen-no-nav` will fix the issue with the blue background not going to the bottom of the screen.

TomBurch committed:
Use grid, apply prettier
